core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -233,13 +252,40 @@
      * @param alloc the {@link ByteBufAllocator} which will allocate the content buffer
      * @param bufferSize the maximum allowed size of the {@link HttpData} buffers
      */
-    static StreamMessage<HttpData> of(Path path, ExecutorService executor, ByteBufAllocator alloc,
+    static StreamMessage<HttpData> of(Path path, @Nullable ExecutorService executor, ByteBufAllocator alloc,
                                       int bufferSize) {
         requireNonNull(path, "path");
-        requireNonNull(executor, "executor");
         requireNonNull(alloc, "alloc");
         checkArgument(bufferSize > 0, "bufferSize: %s (expected: > 0)", bufferSize);
-        return new PathStreamMessage(path, alloc, executor, bufferSize);
+        return of(path, executor, alloc, 0, Long.MAX_VALUE, bufferSize);
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link Path}.
+     * The specified {@code bufferSize} is used to create a buffer used to read data from the {@link Path}.
+     * Therefore, the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
+     * size less than or equal to {@code bufferSize}.
+     *
+     * @param path the path of the file.
+     * @param executor the {@link ExecutorService} which performs blocking IO read.
+     * @param alloc the {@link ByteBufAllocator} which will allocate the content buffer.
+     * @param start the start position of the file to read from; must be non-negative.
+     *              {@code 0} indicates the start of the file.
+     * @param end the end position of the file to read until.
```

Comment:
What do you think of also adding whether the index is exclusive?

```suggestion
     * @param end the end position of the file to read until, exclusive.
```